feat(review) : CRUD 기능 구현

sshkim1013 opened the pull request:
Review에 대한 CRUD 기능 구현하였습니다.
연관 관계 생각하여 ReviewService 클래스에 OrderRepository 필드 선언.

넓은 내용을 알지 못하여 기본적인 CRUD만 최대한 작성하여 PR 올렸습니다.
해당 PR에 대한 리뷰 말씀 부탁드리겠습니다!

감사합니다 다들 파이팅 :)

sshkim1013 committed:
feat(review) : 기능 구현

yeonic reviewed (commented):
주말 새에 시간내어 구현하시느라 고생 많으셨어요!!!
제 눈에 보인 몇가지만 리뷰드리겠습니다~~

- GlobalExceptionHandler에서 @ResponseStatus 지정해줄 것이기 때문에, ResponseStatusException을 던지기보다, 의미에 맞는 생짜 exception을 던지고 handler에서 catch해주는 것이 좋아보입니다.

- 아직 orderRepository가 존재하지 않아 동작하지 않는 것이겠지만, ReviewService의 save는 Review를 생성하는 부분에 findOrder가 repository에서 찾아와 할당한 orders일 것 같은데, 둘이 이름이 다릅니다..! 확인 부탁드려요!

sshkim1013 committed:
Merge branch 'dev' into feat/review

sshkim1013 committed:
refactor(review)
1. Review : Store 연관관계 추가 및 BaseEntity 상속
2. Controller : URL 수정 및 @Valid 추가
3. Service : 중복 검증 로직은 메서드로 분리
4. Repository : 검증 메서드(existByOrders, findByStore... ) 추가
5. RequestDtos : Validation 추가
6. ResponseDtos : createAt, updatedAt 필드 추가

sshkim1013 commented:
어제 담당 튜터님께서 제게 피드백 해주신 내용을 Review 도메인 코드에 최대한 옮겨봤는데, 아직도 많이 미흡한 것 같다는 생각이 듭니다. 
여유되시면 확인 후 리뷰 한 번만 부탁드리겠습니다

sshkim1013 committed:
chore(review) : 객체명 수정(Orders -> Order)

sshkim1013 committed:
Merge branch 'dev' into feat/review

sshkim1013 committed:
refactor(review)

- Review : @Column 어노테이션 제약사항 추가.
- Dto(Response) : @JsonFormat 사용하여 LocalDateTime 포맷팅.
- Controller : 리턴 타입을 ResponseEntity<T> 타입으로 변경.
- Service / Repository : 리뷰 조회 시, findByStoreId가 아닌 findAllByStoreId로 메서드명 변경.
